docs/tutorials/cpp.md

```diff
@@ -1,43 +1,75 @@
+Typically C++ users can build ICU4X by building the `icu_capi` Rust crate, and linking the resultant static library to their C++ application. This crate contains all of the relevant [Diplomat]-generated `extern "C"` declarations.
 
 Using ICU4X in C++ is best demonstrated via the [examples] present in the codebase. For example, [here's an example showing off decimal formatting in ICU4X][decimal-example-code], built with [this Makefile][decimal-example-makefile].
 
 _We are still working on improving the user experience of using ICU4X from other languages. As such, this tutorial may be a bit sparse, but we are happy to answer questions on our [discussions forum] and help you out_
 
-## Building and linking ICU4X
-
-- `cargo build` only supports local crates, so you have to download the published code from crates.io
-    - You can use `cargo clone icu_capi_staticlib@1.0.0` to do this (`cargo install cargo-clone` if the command is not installed)
-- Inside the crate you can use `cargo build` to build the library
-    - Be sure to pass `--release` to get an optimized build
-    - Specify Cargo features with `--features ...`:
-        - `compiled_data` \[default\] to include data (`ICU4XDataProvider::create_compiled()`)
-        - `simple_logger` \[default\] enable basic stdout logging of error metadata. Further loggers can be added on request.
-        - `buffer_provider` for working with blob data providers (`ICU4XDataProvider::create_from_byte_slice()`)
-        - `provider_fs` for loading data from the file system (`ICU4XDataProvider::create_fs()`). This also requires enabling a syntax on the `icu_provider` crate.
-    - Set `CARGO_PROFILE_RELEASE_LTO=true` to enable link-time optimization
-    - Set `CARGO_PROFILE_RELEASE_OPT_LEVEL="s"` to optimize for size
-    - See [cargo profiles](cargo-profiles) for more options
- - Copy the header files from the `icu_capi` crate
-    - If you're using the default crate registry, they will be in `~/.cargo/registry/src/*/icu_capi-1.0.0/cpp/include`
-    - If you cannot find the crate, download it from crates.io
-    - Make sure to use the same version as above
-        - While we selected a version for `icu_capi_staticlib` before, and here we're looking at `icu_capi`, those versions will always be in sync. You can sanity check this by running `cargo pkgid -p icu_capi`.
- - Compile with `g++ -std=c++17 icu/target/release/libicu_capi_staticlib.a -ldl -lpthread -lm`. C++ versions beyond C++17 are supported, as are other C++ compilers.
+## Building ICU4X
+
+After [installing Rust](https://rustup.rs), create a local Rust crate to configure ICU4X:
+
+```shell
+cargo init icu --lib
```

Comment:
nit: I'd call this `icu_wrapper` or something

Reply:
I want the artifact to be called `libicu.a`

Reply:
We already have a Rust library called `icu` though. I don't think it's important for the library to be called `libicu.a`, and it might even be confusing to the user when they have to distinguish between "things they have control over" and "things we provide them".

We should call it `icu_wrapper` or something and mention they can pick whatever name they want.

Reply:
`icu_config`? If only we didn't need a wrapper crate and could just `cargo build-lib`

Reply:
Yeah that's fine. Feels a bit util-y though. `myicu` is also fine.

Reply:
Issue: do _not_ make the artifact `libicu.a`. When linking into C++ there is a real hazard of conflicting or confusing with ICU4C. For shared C libraries, the `icu` namespace belongs to ICU4C, full stop. This is why we use `ICU4X` in all the namespace names in the C++ wrapper.

Also, don't call it `icu_config` because of @srl295's tool named [icu-config(1)](https://linux.die.net/man/1/icu-config)

I think it's somewhat important to make the library name imply that this is a subset of ICU4X with potential customizations to feature flags and things, not stock ICU4X.

Suggestions:

- `libmyicu4x.a`
- `libicu4xcomponents.a`
- `libicu4xslice.a`
- `libicu4xset.a`

Reply:
When you say `icu` namespace, do you mean C++ namespace? If we pass the `.a` explicitly, does the name still matter?

Reply:
I found a way to compile `icu_capi` directly, so it's now `-licu_capi`